File: project/frontend/src/types/dtos/CommentDto.ts
import { CommentType } from "../enums/CommentType.ts";
import type CategoryDto from "./CategoryDto.ts";
import type ProjectDto from "./ProjectDto.ts";
import type LocationDto from "./LocationDto.ts";

export default interface CommentDto {
	id: string | null;
	project?: ProjectDto;
	location: LocationDto;
	categoryId: string | null;
	category?: CategoryDto;
	type: CommentType;
	content: string;
}

File: server/Models/Comments/CommentDto.cs
using server.Models.Categories;
using server.Models.Locations.DTOs;
using server.Models.Projects.DTOs;
using server.Types.Enums;

namespace server.Models.Comments
{
	public record class CommentDto
	{
		public Guid? Id { get; set; }
		public ProjectDto? Project { get; set; }
		public LocationDto? Location { get; set; }
		public Guid? CategoryId { get; set; }
		public CategoryDto? Category { get; set; }
		public CommentType Type { get; set; }
		public string Content { get; set; } = string.Empty;

		// Threading fields
		public Guid? RootCommentId { get; set; }
		public Guid? ParentCommentId { get; set; }
		public int Depth { get; set; } = 0;
		public DateTime? CreatedAt { get; set; }

		// Navigation properties (optional, loaded on demand)
		public ICollection<CommentDto> Replies { get; set; } = [];
	}
}
